heidrun/digcomm_mods.rb

```diff
@@ -0,0 +1,289 @@
+               :as => :coll do
+      title coll
+    end
+
+    ##
+    # If there is no <name> with a <roleTerm>creator</roleTerm>, then the first
+    # name is the creator no matter the <roleTerm>. Any other secondary
+    # <roleTerm> values are contributors. See creator for full instructions for
+    # implementation.
+    #
+    contributor :class => DPLA::MAP::Agent,
+                :each => record.map(&extract_contributors).flatten,
+                :as => :contrib do
+      providedLabel contrib
+    end
+
+    # Combine all sub-fields in the <name> section EXCEPT <affiliation>,
+    # <displayForm>, <description>, and <role>. If there is no <name> with a
+    # <roleTerm>creator</roleTerm>, then the first name is the creator no matter
+    # the <roleTerm>. Any other secondary <roleTerm> values are contributors.
+    creator :class => DPLA::MAP::Agent,
+            :each => record.map(&extract_creator).flatten,
+            :as => :creator do
+      providedLabel creator
+    end
+
+    # <mods:originInfo><mods:dateCreated encoding="w3cdtf" keyDate="yes">
+    # <mods:originInfo><mods:dateIssued encoding="w3cdtf" keyDate="yes">
+    # <mods:originInfo><mods:dateOther encoding="w3cdtf" keyDate="yes">
+    # <mods:originInfo><mods:copyrightDate encoding="w3cdtf" keyDate="yes">
+    date :class => DPLA::MAP::TimeSpan,
+         :each => record.fields(['mods:originInfo', 'mods:dateCreated'],
+                                ['mods:originInfo', 'mods:dateIssued'],
+                                ['mods:originInfo', 'mods:dateOther'],
+                                ['mods:originInfo', 'mods:copyrightDate'])
+                 .match_attribute(:encoding, 'w3cdtf')
+                 .match_attribute(:keyDate, 'yes'),
+         :as => :created do
+      providedLabel created
+      self.begin created.match_attribute(:point, 'start')
+      self.end created.match_attribute(:point, 'end')
+    end
+
+    # <mods:abstract>; <mods:note...>
+    description record.fields('mods:abstract', 'mods:note')
+
+    # <mods:physicalDescription><mods:extent>
+    extent record.field('mods:physicalDescription', 'mods:extent')
+
+    # <mods:genre...>
+    # FIXME: non-DCMIType values from type will be handled in enrichment
+    dcformat record.field('mods:genre')
+
+    identifier record.map(&build_identifier).flatten
+
+    # FIXME: Language code enrichment needed here (language_to_lexvo.rb)
+    #
+    # <mods:language><mods:languageTerm type="text" authority="iso639-2b"
+    # authorityURI="http://id.loc.gov/vocabulary/iso639-2"
+    # valueURI="http://id.loc.gov/vocabulary/iso639-2/code [value]">[text value]
+    language record.field('mods:language', 'mods:languageTerm')
+              .match_attribute(:type, 'text')
+              .match_attribute(:authority, 'iso639-2b')
+              .match_attribute(:authorityURI, 'http://id.loc.gov/vocabulary/iso639-2')
+
+    # FIXME: hierarchicalGeographic is currently coming out as a string like:
```

Comment:
I wasn't sure how best to clean up the content from the hierarchicalGeographic field.  I could do the splitting and reformatting in the mapper, but wasn't sure if this would be better handled by an enrichment?